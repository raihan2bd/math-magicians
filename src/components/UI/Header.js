import React from 'react';
import { NavLink, Link } from 'react-router-dom';

import classes from './Header.module.css';

const Header = () => (
  <header className={classes.header}>
    <nav className={classes.nav}>
      <div className={classes.nav_brand}>
        <h1>
          <Link to="/">Math Magicians</Link>
        </h1>
      </div>
      <ul className={classes.nav_group}>
        <li className={classes.nav_item}>
          <NavLink className={classes.nav_link} to="/">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink className={classes.nav_link} to="/calculator">
            Calculator
          </NavLink>
        </li>
        <li>
          <NavLink className={classes.nav_link} to="/quote">
            Quote
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
);

export default Header;
