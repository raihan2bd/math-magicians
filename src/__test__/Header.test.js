import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Header from '../components/UI/Header';

describe('Test Header component with snapshot', () => {
  test('should match the snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should include the logo', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );

    expect(
      screen.getByRole('heading', { name: 'Math Magicians', exact: false }),
    ).toBeInTheDocument();
  });
  test('should include links', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(screen.getByRole('link', { name: 'Quote' })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Home' })).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: 'Calculator' }),
    ).toBeInTheDocument();
  });
});
