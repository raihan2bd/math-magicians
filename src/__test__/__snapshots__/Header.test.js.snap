// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component with snapshot should match the snapshot 1`] = `
<header
  className="header"
>
  <nav
    className="nav"
  >
    <div
      className="nav_brand"
    >
      <h1>
        <a
          href="/"
          onClick={[Function]}
        >
          Math Magicians
        </a>
      </h1>
    </div>
    <ul
      className="nav_group"
    >
      <li
        className="nav_item"
      >
        <a
          aria-current="page"
          className="nav_link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav_link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="nav_link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
